src/ggml.c

```diff
@@ -10036,11 +10038,10 @@ static void ggml_compute_forward_sub_f32(
     const struct ggml_tensor * src0 = dst->src[0];
     const struct ggml_tensor * src1 = dst->src[1];
 
-    if (params->ith != 0) {
-        return;
-    }
+    GGML_ASSERT(ggml_can_repeat(src1, src0) && ggml_are_same_shape(src0, dst));
```

Comment:
This should remain `assert` because we've already guaranteed these conditions in `ggml_sub_impl` and here we are mostly hinting the developer about what inputs to expect and verify we don't have a bug in the internal `ggml` logic.

In contrast, in `ggml_sub_impl` we have to use `GGML_ASSERT` instead of `assert` because we don't know what the user of the function will pass as input, so we have to check even in non-debug builds

Reply:
Thanks for the fast review!
This should be fixed.